aws/platform/main.tf

```diff
@@ -155,7 +155,7 @@ data "aws_route53_zone" "managed" {
 }
 
 data "aws_s3_bucket_object" "prometheus" {
-  count = var.prometheus_workspace_name == null ? 0 : 1
+  count = var.aws_prometheus_workspace_id != "" ? 0 : (var.prometheus_workspace_name == null ? 0 : 1)
```

Comment:
I think we can remove the older option of accepting a workspace name entirely.

Reply:
This is now added to the new `prometheus_data_source` since the workspace name is needed by service account resource